### PR TITLE
qa/dashboard: ensure node 16 is installed

### PR DESCRIPTION
**Pull request by epuertat**
Fixes: https://tracker.ceph.com/issues/53843
Signed-off-by: Ernesto Puerta <epuertat@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
The affected test is `rados/dashboard/{0-single-container-host debug/mgr mon_election/connectivity random-objectstore$/{bluestore-comp-zlib} tasks/e2e}`; perhaps this PR should be run through Teuthology to make sure the tests are green?

**Comment by djgalloway** [comment]
Can this safely be installed with/over another version of NodeJS?

Why are we installing a deprecated version?

**Comment by epuertat** [comment]
> Can this safely be installed with/over another version of NodeJS?
> 
> Why are we installing a deprecated version?

@djgalloway I simply downloaded the nodejs 12 script since it's the minimum required by [Angular 12](https://www.redcricket.net/en/news-en/compatibility-list-for-angular-angular-cli-and-node-js/) and the one we've been testing in our Jenkinks jobs, but you're right: given 12 is almost EoL, we should update to 16 (the next LTR). Both are supported by CentOS 8 and Ubuntu 20.04. I'll fix it.

File: qa/workunits/cephadm/test_dashboard_e2e.sh [hunk 1]
```diff
@@ -6,10 +6,21 @@ DASHBOARD_FRONTEND_DIR=${SCRIPT_DIR}/../../../src/pybind/mgr/dashboard/frontend
...
 install_common () {
+    NODEJS_VERSION="16"
     if grep -q  debian /etc/*-release; then
         $SUDO apt-get update
-        $SUDO apt-get install -y jq npm
+        # https://github.com/nodesource/distributions#manual-installation
+        $SUDO apt-get install curl gpg
+        KEYRING=/usr/share/keyrings/nodesource.gpg
+        curl -fsSL https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | $SUDO tee "$KEYRING" >/dev/null
+        DISTRO="$(source /etc/lsb-release; echo $DISTRIB_CODENAME)"
```
Comment (epuertat) [inline qa/workunits/cephadm/test_dashboard_e2e.sh:16]: This is a trick not to have to install `lsb-core` (which triggers the install of another 165 packages).

**Comment by epuertat** [comment]
@djgalloway updated, can you please review?

**Review by djgalloway — APPROVED** [review]

**Review by pereman2 — APPROVED** [review]

**Review by nizamial09 — APPROVED** [review]

**Review by alfonsomthd — APPROVED** [review]

**Comment by epuertat** [comment]
> while you're here: you want to move `qa/workunits/cephadm/test_dashboard_e2e.sh ` to `qa/workunits/dashboard/test_dashboard_e2e.sh `?

@sebastian-philipp since @yuriw already validated this PR, I'll keep it this way. We can change it afterwards.

**Comment by ljflores** [comment]
@epuertat posting this for your reference. The dashboard tests that were failing have passed!

http://pulpito.front.sepia.ceph.com/yuriw-2022-01-15_05:47:18-rados-wip-yuri8-testing-2022-01-14-1551-distro-default-smithi/
http://pulpito.front.sepia.ceph.com/yuriw-2022-01-17_17:14:22-rados-wip-yuri8-testing-2022-01-14-1551-distro-default-smithi/

Failures, unrelated:
    https://tracker.ceph.com/issues/45721
    https://tracker.ceph.com/issues/50280
    https://tracker.ceph.com/issues/53827
    https://tracker.ceph.com/issues/51076
    https://tracker.ceph.com/issues/53807
    https://tracker.ceph.com/issues/53842

Details:
    Bug_#45721: CommandFailedError: Command failed (workunit test rados/test_python.sh) FAIL: test_rados.TestWatchNotify.test
    Bug_#50280: cephadm: RuntimeError: uid/gid not found - Ceph
    Bug_#53827: cephadm exited with error code when creating osd. - Ceph - Orchestrator
    Bug_#51076: "wait_for_recovery: failed before timeout expired" during thrashosd test with EC backend.
    Bug_#53807: Dead jobs in rados/cephadm/smoke-roleless{...} - Ceph - Orchestrator
    Bug_#53842: cephadm/mds_upgrade_sequence: KeyError: 'en***'

**Comment by epuertat** [comment]
Thanks a lot for the (detailed) good news, @ljflores ! 🥳

**Pull request merged**